modules/core/test/test_arithm.cpp

```diff
@@ -1495,6 +1496,39 @@ INSTANTIATE_TEST_CASE_P(Core_MinMaxLoc, ElemWiseTest, ::testing::Values(ElemWise
 INSTANTIATE_TEST_CASE_P(Core_CartToPolarToCart, ElemWiseTest, ::testing::Values(ElemWiseOpPtr(new CartToPolarToCartOp)));
 
 
+void testMinMaxLocND(const std::string& relative_path, void (*minMaxLocND)(InputArray, OutputArray, int, bool))
+{
+    const std::string path = findDataFile(relative_path, true);
+    FileStorage file(path, FileStorage::READ);
+    int inputs;
+    file["inputs"] >> inputs;
+
+    for (int i = 0; i < inputs; ++i) {
+        const std::string input = cv::format("input_%d", i);
+        const Mat inp = file[input].mat();
+
+        for (int axis = 0; axis < inp.dims; ++axis) {
+            const std::string output = cv::format("output_%d_%d", i, axis);
+            const cv::Mat gold = file[output].mat();
+
+            cv::Mat out;
+            minMaxLocND(inp, out, axis, false);
+
+            EXPECT_EQ(cv::norm(out, gold), 0.) << input << ", " << output;
+        }
+    }
+};
+
+TEST(Core_MinMaxLocND, ArgMax)
+{
+    testMinMaxLocND("core/argMax.yml", cv::argMax);
```

Comment:
> core/argMax.yml

BTW, currently almost all tests of the "core" module don't require testdata (except filesystem based).
Reference non-optimized implementation is added as part of test code.

Reply:
I know, but this code is already unoptimized, so I wasn't sure how to test it.